packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalBlockNavigationTree as BlockNavigationTree,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { Button } from '@wordpress/components';
+import {
+	useFocusOnMount,
+	useFocusReturn,
+	useInstanceId,
+	useMergeRefs,
+} from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { closeSmall } from '@wordpress/icons';
+import { ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../store';
+
+export default function ListViewSidebar() {
+	const { clientIdsTree, selectedBlockClientIds } = useSelect( ( select ) => {
+		const {
+			__unstableGetClientIdsTree,
+			getSelectedBlockClientIds,
+		} = select( blockEditorStore );
+		return {
+			clientIdsTree: __unstableGetClientIdsTree(),
+			selectedBlockClientIds: getSelectedBlockClientIds(),
+		};
+	} );
+	const { setIsListViewOpened } = useDispatch( editPostStore );
+
+	const { clearSelectedBlock, selectBlock } = useDispatch( blockEditorStore );
+	async function selectEditorBlock( clientId ) {
+		await clearSelectedBlock();
+		selectBlock( clientId );
+	}
+
+	const focusOnMountRef = useFocusOnMount( 'firstElement' );
+	const focusReturnRef = useFocusReturn();
+	function closeOnEscape( event ) {
+		if ( event.keyCode === ESCAPE ) {
+			event.stopPropagation();
+			setIsListViewOpened( false );
+		}
+	}
+
+	const instanceId = useInstanceId( ListViewSidebar );
+	const labelId = `edit-post-editor__list-view-panel-label-${ instanceId }`;
+
+	return (
+		// eslint-disable-next-line jsx-a11y/no-static-element-interactions
+		<div
+			aria-labelledby={ labelId }
+			className="edit-post-editor__list-view-panel"
+			onKeyDown={ closeOnEscape }
+		>
+			<div className="edit-post-editor__list-view-panel-header">
+				<strong id={ labelId }>{ __( 'List view' ) }</strong>
+				<Button
+					icon={ closeSmall }
+					label={ __( 'Close list view sidebar' ) }
+					onClick={ () => setIsListViewOpened( false ) }
+				/>
+			</div>
+			<div
+				className="edit-post-editor__list-view-panel-content"
+				ref={ useMergeRefs( [ focusReturnRef, focusOnMountRef ] ) }
+			>
+				<BlockNavigationTree
+					blocks={ clientIdsTree }
+					selectBlock={ selectEditorBlock }
```

Comment:
Why are these props not something handled inside `BlockNavigationTree` directly: blocks, selectBlock, selectedBlockClientsIs ?

Reply:
Because `BlockNavigationTree` would "normally" be called by `BlockNavigation`, which provides those props:

https://github.com/WordPress/gutenberg/blob/0d1757f3f43a76000b804e561acc7ed6a9c57e8f/packages/block-editor/src/components/block-navigation/index.js#L72-L78

The major difference is that `BlockNavigation` only display the blocks for the currently selected hierarchy (e.g. if you are in a template part, it limits the blocks to those contained in that template part), whereas the persistent list view wants to display all of them.

When I wrote the persistent list view for the Site Editor, I intentionally decided to not touch `BlockNavigation` to avoid affecting the other consumers of the component.

But still, looking at how the list view is used, I can see that most consumers (`edit-post` in this PR, `edit-site`, and the [Navigation block](https://github.com/WordPress/gutenberg/blob/0d1757f3f43a76000b804e561acc7ed6a9c57e8f/packages/block-library/src/navigation/block-navigation-list.js#L32-L40)) use `BlockNavigationTree`, providing their own props.
Only [`edit-widgets` is still using `BlockNavigation`](https://github.com/WordPress/gutenberg/blob/0d1757f3f43a76000b804e561acc7ed6a9c57e8f/packages/edit-widgets/src/components/header/index.js#L95) through the dropdown.

It might be worth it to refactor `BlockNavigation` to reflect this shift, but at the same time, while all sub-components (such as `BlockNavigationTree`) are marked as experimental, `BlockNavigationDropdown` (which uses the main `BlockNavigation`) is exported as stable, and so we would need a deprecation path for this.

Reply:
Looking at the code, it seems the only  difference in `BlockNavigation` is that it only shows the  selected block's tree if it's a container. I'm fine moving that logic into `BlockNavigationTree` with just a flag prop.

Reply:
Since it affects very different areas, I'd do it in a separate PR though, either before or after merging this.